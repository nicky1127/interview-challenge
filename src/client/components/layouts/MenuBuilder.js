import React from "react";
import SideBar from "./SideBar";
import MenuPreview from "./MenuPreview";

const MenuBuilder = () => (
  <div className="container menu-builder">
    <div className="row">
      <div className="col-4">
        <SideBar />
      </div>
      <div className="col-8">
        <MenuPreview />
      </div>
    </div>
  </div>
);

export default MenuBuilder;
