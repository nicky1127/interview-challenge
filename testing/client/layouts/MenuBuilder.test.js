import { render, cleanup } from "@testing-library/react";
import MenuBuilder from "client/components/layouts/MenuBuilder";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});

const state = {
  selectedItemIDs: [],
  items: [{ name: "test", id: 1, dietaries: ["test"] }],
  ldgItems: false,
  ldgSuccItems: false
};
const mockStore = configureMockStore();
const store = mockStore(state);

describe("MenuBuilder", () => {
  it("Snapshot matchs", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MenuBuilder />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("MenuBuilder should render one [data-testid=menuBuilder]", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <MenuBuilder />
      </Provider>
    );
    const container = getByTestId("menuBuilder");
    expect(container).toBeInTheDocument();
  });
});
