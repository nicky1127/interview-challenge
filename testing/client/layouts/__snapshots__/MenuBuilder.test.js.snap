// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBuilder Snapshot matchs 1`] = `
<div
  className="container menu-builder"
  data-testid="menuBuilder"
>
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <div
        data-testid="sideBar"
      >
        <div
          className="filters"
        >
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="Name"
          />
        </div>
        <ul
          className="item-picker"
        >
          <li
            className="sidebar_item item"
            data-testid="sideBarItem"
            onClick={[Function]}
          >
            <h2>
              test
            </h2>
            <p>
              <span
                className="dietary"
                data-testid="sideBarItem_dietary"
              >
                test
              </span>
            </p>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="col-8"
    >
      <div
        data-testid="menuPreview"
      >
        <h2>
          Menu Preview
        </h2>
        <ul
          className="menu-preview"
        />
      </div>
    </div>
  </div>
</div>
`;
